test/compiler/compact.jl

```diff
@@ -0,0 +1,37 @@
+using Core.Compiler: IncrementalCompact, insert_node_here!, complete,
+    NewInstruction, verify_ir, ReturnNode, SSAValue
+
+foo_test_function(i) = i == 1 ? 1 : 2
+
+@testset "IncrementalCompact statefulness" begin
+    ir = only(Base.code_ircode(foo_test_function, (Int,)))[1]
+    compact = IncrementalCompact(ir)
+
+    # set up first iterator
+    x = Core.Compiler.iterate(compact)
+    x = Core.Compiler.iterate(compact, x[2])
+
+    # set up second iterator
+    x = Core.Compiler.iterate(compact)
+
+    # consume remainder
+    while x !== nothing
+        x = Core.Compiler.iterate(compact, x[2])
+    end
+
+    ir = Core.Compiler.complete(compact)
+    @test Core.Compiler.verify_ir(ir) === nothing
+end
+
+# Test early finish of IncrementalCompact
+@testset "IncrementalCompact early finish" begin
+    ir = only(Base.code_ircode(foo_test_function, (Int,)))[1]
+    compact = IncrementalCompact(ir)
+
+    insert_node_here!(compact, NewInstruction(ReturnNode(1), Union{}, ir[SSAValue(1)][:line]))
+    new_ir = complete(compact)
```

Comment:
```suggestion
    new_ir = Core.Compiler.finish(compact)
```
Shouldn't we use `finish`, or `complete` + `Core.Compiler.non_dce_finish!`?

Reply:
Yeah, probably, I got confused by the other test using `complete`. Although this does make me suspicious whether this whole thing shouldn't be in `non_dce_finish!` in the first place. It's already trying to finish the basic block there, it's just doing the wrong one. Let me make the adjustment.